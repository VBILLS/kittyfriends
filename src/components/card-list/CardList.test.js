import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import CardList from './CardList';

describe('it should render shallow snapshot', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Snow',
      username: 'JohnJohn',
      email: 'john@gmail.com'
    }
  ];
  it('matches react-test-renderer snapshot', () => {
    const component = renderer.create(<CardList robots={mockRobots} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('shallow renders & matches snapshot with enzyme', () => {
    const wrapper = shallow(<CardList robots={mockRobots} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
