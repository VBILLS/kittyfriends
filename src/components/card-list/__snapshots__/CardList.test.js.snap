// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render shallow snapshot matches react-test-renderer snapshot 1`] = `
<div
  className="card-list"
>
  <div
    className="card grow"
  >
    <img
      alt="robots"
      src="https://robohash.org/1?size=200x200&set=set4"
    />
    <div
      className="text"
    >
      <h2>
        John Snow
      </h2>
      <p>
        john@gmail.com
      </p>
    </div>
  </div>
</div>
`;

exports[`it should render shallow snapshot shallow renders & matches snapshot with enzyme 1`] = `
<div
  className="card-list"
>
  <Card
    email="john@gmail.com"
    id={1}
    key="0"
    name="John Snow"
  />
</div>
`;
